### PR TITLE
Replace obsolete packages and update to dotnet 8

### PR DESCRIPTION
**Pull request by richardAhz**
Not all files are mentioned, only those with substantial logic or type changes.
Functional tests are not mentioned. 
Common
  Removed Microsoft.Azure.Management.EventGrid (obsolete)
  Removed Microsoft.Azure.EventGrid (obsolete)
  Replaced Newtonsoft.Json 
  
  Using Azure.ResourceManager.EventGrid
  Using System.Text.Json
  
  Code changes
  BaseClass.EventProcessorBase
  Helpers.IAzureEventGridDomainService 
  Helpers.IEventSubscriptionConfiguration
  Helpers.AzureEventGridDomainService 
  Helpers.EventSubscriptionConfiguration
  
  
API
  Removed Microsoft.AspNetCore.Mvc.NewtonsoftJson
  Removed Microsoft.Azure.KeyVault (obsolete) and not used 
  Removed Microsoft.Azure.Services.AppAuthentication (retired) and not used  
  
  Code Changes
  BaseController 
  LoggingMiddleware
  FssEventProcessor
  FssEventValidationAndMappingService
  ScsEventValidationAndMappingService
  
SubscriptionService
  Code Changes
  Services.SubscriptionServiceData

### CODE REVIEW

File: UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.Common/HealthCheck/AzureWebJobHelper.cs [hunk 1]
```diff
@@ -37,9 +38,10 @@ public async Task<HealthCheckResult> CheckWebJobsHealth(WebJobDetails webJob)
 
                 if (response.StatusCode == HttpStatusCode.OK)
                 {
-                    dynamic webJobDetails = JsonConvert.DeserializeObject<dynamic>(await response.Content.ReadAsStringAsync());
-                    string webJobStatus = webJobDetails["status"];
-                    if (webJobStatus != "Running")
+                    string data = await response.Content.ReadAsStringAsync();
+                    JsonNode? webJobDetails = JsonSerializer.Deserialize<JsonNode>(data);   // Rhz Todo
```
Comment (HugoBurgess) [inline UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.Common/HealthCheck/AzureWebJobHelper.cs:42]: There's a Todo left here - does anything still need doing or can you remove it?

File: UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.Common/Helpers/IAzureEventGridDomainService.cs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ﻿using Azure.Messaging;
-using Microsoft.Azure.Management.EventGrid.Models;
+//using Microsoft.Azure.Management.EventGrid.Models;
```
Comment (HugoBurgess) [inline UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.Common/Helpers/IAzureEventGridDomainService.cs:2]: Remove commented out line

File: UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.Common/Helpers/IEventSubscriptionConfiguration.cs [hunk 3]
```diff
@@ -1,10 +1,11 @@
-﻿using Microsoft.Azure.Management.EventGrid.Models;
+﻿using Azure.ResourceManager.EventGrid;
+//using Microsoft.Azure.Management.EventGrid.Models;
```
Comment (HugoBurgess) [inline UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.Common/Helpers/IEventSubscriptionConfiguration.cs:2]: Remove commented out line

File: global.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "sdk": {
+  "sdkxx": {
```
[the remark this answers is not in the document]
Comment (richardAhz) [reply global.json:2]: Done

File: Deployment/src/Modules/Webapp/main.tf [hunk 5]
```diff
@@ -18,7 +18,7 @@ resource "azurerm_app_service" "webapp_service" {
   tags                = var.tags
 
   site_config {
-    windows_fx_version  =   "DOTNET|6.0"
+    windows_fx_version  =   "DOTNET|8.0"
```
Comment (HugoBurgess) [inline Deployment/src/Modules/Webapp/main.tf:21]: GitHub won't let me put this comment in the correct place but there are two other references to "DOTNET|6.0" in this module below.

**Review by HugoBurgess — APPROVED** [review]
Looks good

**Pull request merged**